R/methods-network.R

```diff
@@ -0,0 +1,82 @@
+# Methods for the Network class
+
+
+# Fancy accessors
+setGeneric("getNodes", function(object) standardGeneric("getNodes"))
+setMethod("getNodes", "Network", function(object) object@nodes)
+setGeneric("getLinks", function(object) standardGeneric("getLinks"))
+setMethod("getLinks", "Network", function(object) object@links)
+setGeneric("getLinkColorScheme", function(object) standardGeneric("getLinkColorScheme"))
+setMethod("getLinkColorScheme", "Network", function(object) object@linkColorScheme)
+# No setters! Once created, a network should only be updated via network methods
```

Comment:
Even changing the color scheme?

Reply:
yes. Cannot change the color scheme without going through a network method. Otherwise i could just say setColorScheme to brown and the attribute would change but the node colors won't change. I'd rather the function be real `colorNodes(network, colorScheme)` and `colorLInks(network, colorScheme)` functions so it's more clear they are really doing a lot, not just setting a property.

Reply:
I think this moves me further into color as a method param territory